lib/cn/cn_tree.c

```diff
@@ -1965,20 +1961,262 @@ static merr_t
 cn_subspill_apply(struct subspill *ss)
 {
     struct cn_compaction_work *w = ss->ss_work;
-    merr_t err;
+    merr_t err = 0;
+    bool delete_node = false;
+    struct cn_tree_node *outnode;
+
+    if (!ss->ss_added)
+        return 0;
+
+    if (ss->ss_is_zspill) {
+        struct kvset_list_entry *le;
+
+        err = cn_move(w, w->cw_node, ss->ss_zspill.zsp_src_list,
+                      w->cw_kvset_cnt, delete_node, ss->ss_node);
+
+        w->cw_output_nodev = &outnode;
+
+        if (err) {
+            uint i = 0;
+
+            list_for_each_entry(le, &w->cw_node->tn_kvset_list, le_link) {
```

Comment:
Seems like this loop should go from w->cw_mark for w->cw_kvset_cnt kvsets in reverse order?

Reply:
yes, good point